third_party/amd/lib/TritonAMDGPUTransforms/OptimizeDotOperands.cpp

```diff
@@ -205,6 +205,69 @@ class ReuseShmemForDirectAndTransposedUse : public OpRewritePattern<LoadOp> {
   triton::AMD::ISAFamily isaFamily;
 };
 
+template <typename OpTy>
+class AllocSharedMemForUpcastedScales : public OpRewritePattern<OpTy> {
+public:
+  using OpRewritePattern<OpTy>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(OpTy op,
+                                PatternRewriter &rewriter) const override {
+    auto forOp = op->template getParentOfType<scf::ForOp>();
+    if (!forOp)
+      return rewriter.notifyMatchFailure(op,
+                                         "Don't alloc lds outside for loop");
+
+    BackwardSliceOptions options;
+    options.omitBlockArguments = true;
+    SetVector<Operation *> slice;
+    (void)getBackwardSlice(op.getOperand(1), &slice, options);
+    tt::LoadOp loadOp;
+    unsigned cnt = 0;
+    bool hasAllocatedLDS = false;
+    for (auto &op : slice) {
+      if (isa<tt::LoadOp>(op)) {
+        loadOp = dyn_cast<tt::LoadOp>(op);
+        cnt++;
+      } else if (isa<ttg::LocalAllocOp, ttg::LocalLoadOp>(op)) {
```

Comment:
Isn't checking `ttg::LocalLoadOp` enough? If we have `ttg::LocalAllocOp` but it's not used in this op chain that's also fine?

Reply:
Didn't know other usecases. Yes checking `ttg::LocalLoadOp` is enough.